textract2page/convert_aws.py

```diff
@@ -201,6 +217,21 @@ def convert_file(json_path: str, img_path: str, out_path: str) -> None:
             awsgeometry = TextractPolygon(line_block["Geometry"]["Polygon"])
         else:
             awsgeometry = TextractBoundingBox(line_block["Geometry"]["BoundingBox"])
+
+        # wrap lines in separate TextRegions to preserve reading order
+        # (ReadingOrder references TextRegions)
+        line_region_id = f'line-region-{line_block["Id"]}'
+        pagexml_text_region_line = TextRegionType(
+        Coords=CoordsType(
+            points=points_from_awsgeometry(awsgeometry,
+                                           pil_img.width,
+                                           pil_img.height)
+        ),
+        id=line_region_id,
+        )
+        pagexml_text_region.add_TextRegion(pagexml_text_region_line)
```

Comment:
It's of course possible in PAGE-XML to use recursive regions, but not all tools (including not all OCR-D processors) actually iterate below the top level. So at this point I'd rather add the line-regions directly to the page.